apiserver/facades/agent/uniter/uniter.go

```diff
@@ -153,7 +153,7 @@ func (u *UniterAPI) EnsureDead(ctx context.Context, args params.Entities) (param
 			continue
 		}
 
-		_, err = u.removalService.RemoveUnit(ctx, unitUUID, false, time.Duration(0))
+		_, err = u.removalService.RemoveUnit(ctx, unitUUID, false, false, time.Duration(0))
```

Comment:
`EnsureDead` does not actually call this in 3.6; I will fix in a follow-up.

Reply:
This was changed for 3.6, so shouldn't be be following that?

Reply:
> In fact it makes it Dead even if Alive!

We should never do that. It breaks all assumptions about the life cycle of a unit.

Reply:
But it's central behaviour in 3.6, which we are trying broadly to replicate. Changing this pulls the rug out from too many assumptions.

Reply:
I'd say that's a bug in 3.6 that should be fixed. I know I'm being overly pedantic, but this is the reason why removal is harder than it should be (although admittedly its a lot better in 4) , because we are taking short cuts and not enforcing the core concepts.

I'm not going to push this any further.